samcli/commands/build/build_context.py

```diff
@@ -236,20 +236,20 @@ def run(self):
 
         self._stacks = self._handle_build_pre_processing()
 
-        # boolean value indicates if mount with write or not, defaults to READ ONLY
-        mount_with_write = False
-        if self._use_container:
-            if self._mount_with == MountMode.WRITE:
-                mount_with_write = True
-            else:
-                # if self._mount_with is NOT WRITE
-                # check the need of mounting with write permissions and prompt user to enable it if needed
-                mount_with_write = prompt_user_to_enable_mount_with_write_if_needed(
-                    self.get_resources_to_build(),
-                    self.base_dir,
-                )
-
         try:
+            # boolean value indicates if mount with write or not, defaults to READ ONLY
+            mount_with_write = False
+            if self._use_container:
+                if self._mount_with == MountMode.WRITE:
+                    mount_with_write = True
+                else:
+                    # if self._mount_with is NOT WRITE
+                    # check the need of mounting with write permissions and prompt user to enable it if needed
+                    mount_with_write = prompt_user_to_enable_mount_with_write_if_needed(
+                        self.get_resources_to_build(),
```

Comment:
Question, the call to `self.get_resources_to_build()` makes another call that looks like it could raise a `ResourceNotFound` exception, should this also be caught?

Reply:
Good catch. It was already called in `ApplicationBuilder`'s constructor and we were printing the full stack trace. Now we are gracefully ending the execution there.